src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -225,6 +233,17 @@ public unsafe static bool TryGetPublicKey(ImmutableArray<byte> blob, out Immutab
                 // The RSA magic key must match the blob id
                 if (type == PrivateKeyBlobId && magic == RSA2)
                 {
+                    RSAParameters privKey = default;
+                    privKey.Exponent = expBytes;
+                    privKey.Modulus = pubKeyData;
+                    privKey.P = blobReader.ReadBytes(modulusLength / 2);
```

Comment:
I *think* we made this work on all OSes:

```C#
using (RSA rsa = new RSACryptoServiceProvider())
{
    rsa.ImportCspBlob(blobButAsAStandardArray);
    RSAParameters privateParams = rsa.ExportParameters(true);
    RSAParameters publicParams = rsa.ExportParameters(false);
}
```

If you don't want to have to parse CAPI blobs yourself.